### ISSUE TITLE
Prevent errors when running script

### ISSUE THREAD

**Issue by ghost** [kamilion/customizer#153]
This is follow up from issue #141 that had reintroduced install script for "old stable". While it works well for intended use (to make installation easier), some users may be aware of known issues when using the script i.e. `INSTALL` and also related to `Makefile`.

The following known issues have been identified.

**A. Install script will fail when run from other than Home directory**

Due to `cd` command is used in the script, in case users are picky and insist to run from other directory, install script fail with dpkg error as shown in the following.

    dpkg: error processing archive customizer*.deb (--install):
     cannot access archive: No such file or directory
    Errors were encountered while processing:
     customizer*.deb

I thought of two ways to solve: (a) support other directories by using `cd ..` command to change to upper level of current directory, or (b) prevent running script if current directory is not Home directory

Here, I'd prefer (b) because Home directory is the default directory across all known system i.e. when opening Terminal or File manager, it works well since permission is always set to the current user, and no need to update "First guide" to amend the command for this trivial issue.

**How to implement:** add conditional test in new function `show_warn` after the first call of `show_msg` and if current directory does not match with Home directory, assign `KEY=q` then script won't traverse into `while` loop, then second call of `show_msg` will show `Finished now` then exit.

```
# FUNCTIONS FOR DISPLAY:
...
function show_warn {
    if [ "$PWD" != "$HOME" ]
        then echo "[!] Script is not run from $HOME" && KEY="q"
    fi
}
...
# BEGIN SCRIPT:
show_msg
show_warn
while [ "$KEY" != q ]
...
```

The expected result would be:

    ~$ cd test
    ~/test$ bash $HOME/TEST
    Started INSTALL script
    Use this script for Customizer "old stable"
    [!] Script is not run from /home/trusty
    Finished now

The new function `show_warn` could include future additions for other warnings to prevent any error when running the script, should there be any.

**B. `make deb` will generate misleading 'fatal' error**

There was a time that `make deb` had bothered me after one user noticed that if `git` was not installed, this will result in "not found" error (from unresolved comments left in issue #54 almost two years ago, when `git` was not included in First guide yet)

> I saw two instances of "/bin/sh: 1: git: not found" so I thought what harm in apt-get install git. After I did so. the program installed without further issue

Similarly, even `git` is installed, 'fatal' errors will be shown if `.git` directory doesn't exist (this is normal if user is running `make deb` from non-cloned directory)

    ...
    fatal: Not a git repository (or any of the parent directories): .git
    sed -e 's|@VERSION@|4.1.3 (old stable)|' -e 's|@PREFIX@|/usr|g' \
       	src/main.py.in > src/main.py
    fatal: Not a git repository (or any of the parent directories): .git
    sed -e 's|@PREFIX@|/usr|' -e 's|@ELEVATOR@|pkexec|' \
       	data/customizer.menu.in > data/customizer.menu
    ...

The root cause is found in the first two lines of `Makefile`:

 ```
 VERSION = 4.1.4 ($(GIT_VERSION))
 GIT_VERSION = $(shell $(GIT) rev-parse --short HEAD || echo "old stable")
 ```

`Makefile` was likely optimized by original developer to easily parse the unique commit number from `git` while developing Customizer in a cloned directory; example result of unique commit number is shown in the following.

    $ grep 'aboutLabel' Customizer.2/debian/customizer/usr/sbin/customizer-gui
    ui.aboutLabel.setText('<b>Customizer 4.1.4 (917dde6)</b>')

Otherwise, normal installation will result as "(old stable)" in the created binary; this however, would present misleading 'fatal' error when creating binary and probably went unnoticed or ignored since end result works fine.

    $ grep 'aboutLabel' Customizer.2/debian/customizer/usr/sbin/customizer-gui
    ui.aboutLabel.setText('<b>Customizer 4.1.4 (old stable)</b>')

Here, I'd prevent the misleading 'fatal' error by assigning `GIT_VERSION` using [conditional syntax](https://www.gnu.org/software/make/manual/html_node/Conditional-Syntax.html#Conditional-Syntax) and [wildcard function](https://www.gnu.org/software/make/manual/html_node/Wildcard-Function.html#Wildcard-Function) to filter target directory for GNU Make.

**How to implement:** ensure that if `.git` directory exist, then run the `git` command, otherwise run `echo "old stable"`; therefore the first two lines will be revised and replaced by the following lines of code, with `GIT_VERSION` renamed to `REVISION`.

```
VERSION = 4.1.4 ($(REVISION))
REVISION = 
ifeq ($(wildcard .git), .git)
REVISION = $(shell $(GIT) rev-parse --short HEAD)
else
REVISION = $(shell echo "old stable")
endif
```

Note that there is no spaces or tabs before the conditional syntax. This is found to be proper in makefile, which is different from Bash or Python.

The expected result from this changes is, no more 'fatal' errors shown while running `make deb`, which may mislead users.

**This can be fixed by several direct commits to `master` branch, probably sooner than other existing issues in milestone 4.1.5.**

**Comment by ghost** [comment]
While known issues have been prevented in recent commits, I am unsure if the script will be error-free until I improvise and automate install script.

This issue will only be closed when issue #158 is resolved.

**Comment by ghost** [comment]
This issue shall be closed, because "old stable" will be deprecated. Consider as resolved i.e. misleading fatal error by git command was resolved earlier. No more known issues.
